python/az/aro/azext_aro/tests/latest/unit/test_validators.py

```diff
@@ -1276,3 +1277,51 @@ def test_validate_cluster_identity(test_description, namespace, expected_excepti
 
     if expected_identity is not None:
         assert (expected_identity == namespace.mi_user_assigned)
+
+
+test_validate_upgradeable_to_data = [
+    (
+        "should not raise any Exception because namespace.upgradeable_to is empty",
+        Mock(upgradeable_to="", client_id=None, client_secret=None),
+        None, None
+    ),
+
+    (
+        "should raise RequiredArgumentMissingError Exception because namespace.client_id is present",
+        Mock(upgradeable_to="4.14.5", client_id="client_id_456", client_secret=None),
+        RequiredArgumentMissingError, '--client-id must be not set with --upgradeable-to.'
+    ),
+
+    (
+        "should raise RequiredArgumentMissingError Exception because namespace.client_secret is present",
```

Comment:
Just for completeness, should we add a test case for when --client-id, --client-secret, and --upgradeable-to are all present?